lib/salus/scanners/semgrep.rb

```diff
@@ -114,7 +115,13 @@ def run
                 "- #{msg}\n" \
                 "\t#{hit_to_string(hit, base_path)}"
               end
+              hit_id = if hit['check_id'] == '-' # id not specified by user
```

Comment:
`check_id` is always present right ?

Reply:
No.  

If you directly use a semgrep config, then the `check_id` is always there in the semgrep config.
```yaml
# salus.yaml
scanner_configs:
  Semgrep:
    matches:
      - config: semgrep_config_1.yaml
        forbidden: true
```

```yaml
rules:
  - id: eqeq-always-true
    patterns:
      - pattern: $X == $X
      - pattern-not: 0 == 0
    message: "$X == $X is always true"
    languages: [python]
    severity: ERROR
```

But, you can also directly specify the pattern in the salus.yaml, and there would be no ids.  Like
```yaml
scanner_configs:
  Semgrep:
    matches:
      - pattern: $X == $X
        message: Useless equlity check
        language: python
        forbidden: true
```

Reply:
Was just wondering if we want to enforce ids but i am okay with using if present if not hashing it.